infra/logging/src/main/java/com/simprints/infra/logging/Simber.kt

```diff
@@ -119,13 +138,51 @@ object Simber {
      * 1 kB in size."
      */
     fun tag(tag: String, isUserProperty: Boolean = false): Simber {
-        if (isUserProperty)
-            Timber.tag(USER_PROPERTY_TAG + tag)
-        else
-            Timber.tag(tag)
+        var conformingTag = tag
+
+        // Firebase Analytics requires tag to be no longer than 24 characters
+        var maxLength = 24
+        if (isUserProperty) {
+            conformingTag = USER_PROPERTY_TAG + conformingTag
+            // Firebase Analytics requires user property tag to be no longer than 40 characters
+            maxLength = 40
+        }
+        conformingTag = ensureMaxLength(conformingTag, maxLength)
+        conformingTag = ensureCharactersAreValid(conformingTag)
+
+        Timber.tag(conformingTag)
         return Simber
     }
 
+    /*
+    * Check that tag complies with Firebase Analytics requirements:
+    * Name must consist of letters, digits or _ (underscores).
+     */
+    private fun ensureCharactersAreValid(tag: String): String {
+        var validTag = tag
+        if (validTag.contains(Regex("[^a-zA-Z0-9_]"))) {
+            // Throw an exception in debug but replace invalid characters in other modes
+            if (BuildConfig.DEBUG) {
+                throw IllegalArgumentException("Tag must consist of letters, digits or _ (underscores).")
+            } else {
+                validTag = validTag.replace(Regex("[^a-zA-Z0-9_]"), "_")
+            }
+        }
+        return validTag
+    }
+
+    private fun ensureMaxLength(message: String, maxLength: Int): String {
```

Comment:
My first instinct when seeing this name is that it pads message always to be precisely maxLength. :D Maybe something like `limitLength(message, max)`?